### ISSUE TITLE
5.0.0 Release Discussion

### ISSUE THREAD

**Issue by joshxg** [openmsupply/mobile#2445]
## RELEASE VERSION

v5.0.0

### SUMMARY

This release will be primarily focused on closing off some customisations for dispensing functionality to mobile. This will include being able to look up all patients from the mSupply server - and handling dispensing to patients from many stores. There are also a few outstanding customisations which will need to be completed.

The development deadline is 17th of April with the 20th of April being the target release candidate date. Release aimed for as soon after that as possible, depending on testing capacity.

If anyone has any other features/bugs/refactors they think need to be in this release, or think we should discuss feel free to comment!

### Features

- [x] Portuguese language support : #2115 
- [x] Spanish language support : #2464  
- [x] Add program column to requisitions and stocktakes: #2386
- [x] Integration of a patient lookup API from mSupply: #2446 
- [x] Supporting incoming customer credits: #2451 
- [x] Remove the reason column from stocktake screen: #2382 

### Refactors/Enhancements

- [x] Dependency upgrades: #2447 
- [x] Using prefs instead of custom data: #2516 
- [x] Ability to hide snapshot quantity column: #2509 
- [x] Add a units column to stocktakes: #2508 
- [x] Add bugsnag to fetch hook: #2632
- [x] Dispensary record lookup form should add wildcards by default: #2713
- [x] Customer credits allow for setting negative quantities: #2756
- [x] Remove unnecessary bugsnagging from fetch hook: #2810.
- [x] Update dispensary UI: #2711.
- [x] Update patient history modal columns: #2718.

### Bugs

- [x] French Translation Request: #2390.
- [x] French Translation Request: #2389.
- [x] Untranslated string: #2449.
- [x] Example comment in translation.js: #2409.
- [x] Long Patient names cause overlapping issues: #2467.
- [x] Non-centered modal titles: #2476.
- [x] Edit sync URL warning showing on a single line: #2478.
- [x] French translations: #2479.
- [x] Supplier invoice page batch column unsortable/misaligned: #2482.
- [x] Stop unfinalised prescriptions from syncing  #2494.
- [x] Last sync date is null: #2666.
- [x] Cannot cancel when confirming an insurance policy for non-local patient: #2717.
- [x] Cash on prescriptions does not effect cash register balance: #2720.
- [x] New patient form is disabled: #2721.
- [x] Prescription form incorrect header title: #2735.
- [x] Looking up patient by policy number is not working: #2749.
- [x] Hardware back button is double handled: #2752.
- [x] Dispensary lookup records not resetting on empty search: #2755.
- [x] Lookup modal not showing error message when server is not accessible: #2757.
- [x] Crash on customer requisition use required/suggested quantity: #2758.
- [x] Menu page badges do not refresh on back navigation: #2764.
- [x] Program column not sortable: #2773.
- [x] Stocktake batches disabled for finalised stocktakes: #2779.
- [x] Crash on deleting new records: #2780.
- [x] Incorrect patient date of birth in patient lookup: #2790.
- [x] Supplier and customer invoice inconsistent search filter text: #2796.
- [x] Invalid date error on sync: #2800.
- [x] Customer credit sync error: #2801.
- [x] Adding items from master list to supplier requisition causes crash: #2802.
- [x] Icons for sorting requisition number columns are not visible in supplier requisitions page: #2804.
- [x] Options to select months of stock are too small: #2805.
- [x] Supplier and reason dropdown icons are not disabled for finalised stocktakes: #2812.
- [x] Supplier invoice page missing supplier credit page info: #2821.
- [x] Customer requisition indicators not updating: #2825.

### DOCUMENTATION

- [ ] Portuguese language support
    - [ ] DRAFT:
    - [ ] FINAL:
- [ ] Add a Programme column to a few pages
    - [ ] DRAFT:
    - [ ] FINAL:
- [ ] Integration of a patient lookup API
    - [ ] DRAFT:
    - [ ] FINAL:
- [ ] Incoming customer credits
    - [ ] DRAFT:
    - [ ] FINAL:
- [ ] Using prefs instead of custom data
    - [ ] DRAFT:
    - [ ] FINAL:

### TESTING

- [x] Customer requisitions page: #1243.
- [x] Customer requisition page: #1244.
- [x] Supplier requisitions page: #1238.
- [x] Supplier requisition page: #1239.
- [x] Customer invoices page: #1233. 
- [x] Customer invoice page: #1234.
- [x] Supplier invoices page: #1237.
- [x] Supplier invoice page: #1236.
- [x] Stock page: #1245.
- [x] Stocktakes page: #1240.
- [x] Stocktake manage page: #1241.
- [ ] Stocktake edit page: #1242.
- [x] Dispensary page: #2106.
- [x] Lookup modal: #2734.
- [x] Prescription page: #2108.
- [x] Cash register: #2182.
- [x] Supplier credits: #2336.
- [x] Indicators: #1935.

**Comment by wlthomson** [comment]
@joshxg @mark-prins Some ideas/questions to get the ball rolling:

- Do we have a basic idea of the key customisations involved? I know @mark-prins is working on confirming the work with IC, but might be helpful to document the proposed work in this issue and mark them as blocked by funding (can update as things get confirmed)? Might help just to get a scale of the work.

- For the patient API, I think maybe best to move the discussion to @josh issue and use that as an epic (https://github.com/openmsupply/mobile/issues/2446)? Thinking we will probably need an epic separate from the release discussion to discuss/design the UI, logic, etc. and also to keep track of related issues on the desktop end?

- Do we have a hard deadline for this work from IC? If not, do we have a rough deadline that we want to work towards?

**Comment by joshxg** [comment]
The issues regarding customisations would be: 
- #2390 
- #2389 
- #2386 
- #2383 

There are also quite a few which have not been confirmed as issues to action:
- #2224 
- #2225 
- #2388 
- #2387 
- #2385 
- #2382 
- #2229 
- #2225 

I think we'll be able to lock down all these issues in within the next day or two - whether we are doing them or not.

There is also the patient API: https://github.com/openmsupply/mobile/issues/2446 and also we need a way to limit the amount of credit a patient can use https://github.com/sussol/msupply/issues/5698 (no mobile issue yet) - which might need to be a customisable amount - maybe a pref from mSupply.

Seems like a lot but I don't think it's too crazy - most of the issues are quite simple. Especially if we still aren't accepting patient history and integrating transactions from other stores.

**Comment by joshxg** [comment]
There are also a few tidy-up ish issues from the dispensing feature which I think we should try and work on with this release: https://github.com/openmsupply/mobile/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aissue+label%3A%22Module%3A+dispensary%22+-label%3A%22Priority%3A+unconfirmed+issue%22+

These are mainly refactors, which would be good to get done sooner rather than later.

------- 

We also have the portugese translations in #2337 which I put off into 4.0.0 - I'm not sure how important it is to have these into a release? Maybe they can wait until 4.1.0 - might need to check with craig on that one.

-------

We also are starting work again on Vaccines - which will take a bit longer than this release - a stab in the dark at this release would be about a month? Would have to confirm in the next couple of days. So we will be running vaccines development in parallel with this release in a separate feature branch

--------

last but not least is that we have not had a dependencies upgrade in quite a few months, and we're definitely due! #2447 - should probably be priority #1 i think!

**Comment by mark-prins** [comment]
thanks @joshxg - agree on all counts. 
dependencies and tidy up / refactoring would be great to get done first while we plan the patient API. The CI issues are mostly confirmed as in or out now, so I'll close the 'out' ones.

**Comment by joshxg** [comment]
Have updated the description with some issues - I suppose just a date! Any thoughts on when we should be trying to deliver the patient api feature @mark-prins ?

**Comment by joshxg** [comment]
Also have chucked https://github.com/openmsupply/mobile/issues/2451 into this release which has just come up. I think with the addition of supplier credits in mobile, it's become high priority, even though it has technically always been the case!

**Comment by joshxg** [comment]
Have done #2453 for dependency upgrades. This doesn't include `react-navigation`. They've had a major version release which they've been working on for a couple of years, I think. It's significant changes and it looks like it might take a couple of days to upgrade - maybe. Maybe not so bad, I'll check it out properly and report back!

**Comment by joshxg** [comment]
@mark-prins @wlthomson 

It looks like the date for desktop is going to be between the 4-6th of April. So maybe we should try to release this on the 6th? That would mean probably wrapping up development by the 30th March, giving a week of testing. Basically 3 working weeks of development + 1 week of testing from today?

**Comment by joshxg** [comment]
Have also added in Spanish language support which is already done.. Thanks @diegoevangelisti !!

I think @nmadruga mentioned we need to have some sort of blog post with the portugesueuse language support release.. so maybe the same thing for spanish? I dunno - will let you guys sort it out? Looking at about a month for the release so plenty of time!

**Comment by diegoevangelisti** [comment]
> I think @nmadruga mentioned we need to have some sort of blog post with the portugesueuse language support release.. so maybe the same thing for spanish? I dunno - will let you guys sort it out? Looking at about a month for the release so plenty of time!

Sure, my plan is to talk with @nmadruga on Monday about docs related with Spanish. Thanks for bringing this up!

**Comment by wlthomson** [comment]
@joshxg Planning to release patch `v4.0.1` today, so will be looking at focusing on this from tomorrow onwards. Assuming 30th March, that gives a tidy development window of 3 weeks, which I think is a good time-frame. 

TLDR; +1 for setting 30th March as the development cut-off date.

**Comment by wlthomson** [comment]
Update:

Have added IC phase 2 issues: #2508, #2509.

**Comment by wlthomson** [comment]
Update: @joshxg @mark-prins as discussed, have bumped next version to major rather than minor due to possibility of backwards incompatible API changes.

**Comment by andreievg** [comment]
@wlthomson @joshxg 
Just got a go ahead, custom data to preferences is happening in 4.10: https://github.com/sussol/msupply/issues/5762

So we would like this one in 5
https://github.com/openmsupply/mobile/issues/2516

**Comment by joshxg** [comment]
cool - also a bug with insurance I've added to the list: https://github.com/openmsupply/mobile/issues/2523

**Comment by wlthomson** [comment]
Update:

Target release candidate pushed back to 20th as desktop release has been bumped. Hoping to get a release that week, but will depend on testing capacity and how many people will still be busy with `v10` and `v11`.

**Comment by wlthomson** [comment]
Update: 

`v5.0.0-RC1` released. Testing to start this week with a pretty quick turnaround to get a release before `v6.0.0` starts moving!

**Comment by wlthomson** [comment]
Update:

All test templates are updated and ready for testing: https://github.com/openmsupply/mobile/issues?q=is%3Aopen+is%3Aissue+label%3A%22Test+Plan+Template%22+milestone%3A5.0.0.

**Comment by wlthomson** [comment]
Update: 

Thanks for everyone testing! A few more issues have popped up than expected. Most fixed, a few remain to do which I will prioritise this week. On track for a tidy release date of March 1st :tada:.

**Comment by wlthomson** [comment]
Update: 

Have updated description with a list of template issues. Hopefully helps give a good idea of the amount of testing done/left to do! Thanks again to everyone testing!

**Comment by wlthomson** [comment]
Update: 

`v5.0.0-RC1` release for testing :tada:.

Hopefully can get the remaining template issues done by end of the week, would be good to get `v5.0.0` released by Monday 11th!

**Comment by wlthomson** [comment]
Update:

`v5.0.0` released 🎉.

Any existing `v5` issues will be bumped to `v5.1` release issue (https://github.com/openmsupply/mobile/issues/2836).

Thanks to all the testers and reviewers!

**Comment by diegoevangelisti** [comment]
> Update:
> 
> `v5.0.0` released 🎉.
> 

Awesome! Great job @wlthomson and team

**Comment by wlthomson** [comment]
Closing! :tada: :tada: :tada: 

For any and all remaining or new `v5` issues/discussion, see https://github.com/openmsupply/mobile/issues/2836.

**Comment by wlthomson** [comment]
Update: 

`v5.0.1` patch released. 

For more info, see #2853.

**Comment by wlthomson** [comment]
Update:

`v5.0.2` patch released.

For more info, see #2854.

**Comment by wlthomson** [comment]
Update: 

`v5.0.3` patch released.

For more info, see #2922.
